### ISSUE TITLE
The training loss of id_loss descends when using random erasing data augmentation,but...

### ISSUE THREAD

**Issue by jishanshanss** [ifzhang/FairMOT#227]
In my training , I only trained MOT17 dataset, the original loss is(mot17.json, ,batch_size = 12,  lr_step=20,27):  
'''
2020-09-03-09-07: epoch: 1 |loss 12.723625 | hm_loss 1.376580 | wh_loss 2.511866 | off_loss 0.227324 | id_loss 5.791194 | time 9.250000 | 
2020-09-03-13-35: epoch: 30 |loss 4.259933 | hm_loss 0.329579 | wh_loss 0.948180 | off_loss 0.188468 | id_loss 2.632285 | time 9.116667 | 
'''
And then using random erasing ,i got the following （lr_step=20,27)）:
'''
2020-09-02-10-05: epoch: 1 |loss 10.485511 | hm_loss 1.262890 | wh_loss 2.724560 | off_loss 0.228408 | id_loss 4.648225 | time 10.100000 | 

2020-09-02-13-14: epoch: 20 |loss -0.000416 | hm_loss 0.394108 | wh_loss 1.123665 | off_loss 0.198913 | id_loss 0.171012 | time 10.116667 | 

2020-09-02-14-53: epoch: 30 |loss -0.157708 | hm_loss 0.374578 | wh_loss 1.073495 | off_loss 0.196907 | id_loss 0.115822 | time 10.100000 | 
'''

And I retrained  the data as you commented  in #201, set lr_step=20,and using the random erasing data augmentation too, the loss is normal:
'''
2020-09-02-19-11: epoch: 20 |loss 0.276070 | hm_loss 0.361395 | wh_loss 0.965656 | off_loss 0.190672 | id_loss 0.167447 | time 9.183333 | 

2020-09-02-20-44: epoch: 30 |loss 0.125276 | hm_loss 0.344842 | wh_loss 0.962393 | off_loss 0.190527 | id_loss 0.161323 | time 9.216667 | 
'''
Q1:As you see , the loss of id_loss is lower，but  in the case of  lr_step=20,27, from epoch 20,the total loss became negative, I find the defination of the loss in your code ,and it is so complex, also i see your commentation in #38 ,but i am still not sure if it is normal?
Q2: The test data set has no gt.txt ， I do not know how to compute the MOTA or MOTP to confirm if it is useful to using random erasing method,  I cannot sign up the https://motchallenge.net/ ,Is there any other way to get the MOTA、MOTP results?

**Comment by jishanshanss** [comment]
One more question：Did you use random erasing or other data augmentation methods in your training?

**Comment by ifzhang** [comment]
1. Negative loss is normal because we use uncertainty loss.
2. The MOTA of the test set can only be obtained from the mot challenge webpage.
3. I did not use other data augmentation in my training.

**Comment by jishanshanss** [comment]
> 1. Negative loss is normal because we use uncertainty loss.
> 2. The MOTA of the test set can only be obtained from the mot challenge webpage.
> 3. I did not use other data augmentation in my training.


Thank you very much for your reply！！！
